internal/services/securitycenter/advanced_threat_protection_resource.go

```diff
@@ -88,6 +88,35 @@ func resourceAdvancedThreatProtectionCreateUpdate(d *pluginsdk.ResourceData, met
 		return fmt.Errorf("updating Advanced Threat protection for %q: %+v", id.TargetResourceID, err)
 	}
 
+	deadline, _ := ctx.Deadline()
+
+	// the API appears to be eventually consistent, tracked on https://github.com/Azure/azure-rest-api-specs/issues/25232
+	stateConf := &pluginsdk.StateChangeConf{
+		Pending: []string{"diff"},
+		Target:  []string{"consistent"},
+		Refresh: func() (result interface{}, state string, err error) {
+			resp, err := client.Get(ctx, id.TargetResourceID)
+			if err != nil {
+				return resp, "error", err
+			}
+			if atpp := resp.AdvancedThreatProtectionProperties; atpp != nil {
+				respEnabled := atpp.IsEnabled != nil && *atpp.IsEnabled
+				if respEnabled == d.Get("enabled").(bool) {
+					return resp, "consistent", nil
+				} else {
+					return resp, "diff", nil
+				}
+			}
+			return resp, "error", fmt.Errorf("Properties was nil")
+		},
+		MinTimeout: 1 * time.Minute,
+		Timeout:    time.Until(deadline),
```

Comment:
presumably if this is eventually consistent we also want to wait for this to be hit a repeated value of times (e.g. 3 times in a row?)